airflow/contrib/hooks/gcp_vision_hook.py

```diff
@@ -354,3 +307,137 @@ def _get_autogenerated_id(response):
+    def delete_reference_image(
+        self,
+        location,
+        product_id,
+        reference_image_id,
+        project_id=None,
+        retry=None,
+        timeout=None,
+        metadata=None,
+    ):
+        """
+        For the documentation see:
+        :py:class:`~airflow.contrib.operators.gcp_vision_operator.CloudVisionReferenceImageCreateOperator`
+        """
+        client = self.get_conn()
+        self.log.info('Deleting ReferenceImage')
+        name = ProductSearchClient.reference_image_path(
+            project=project_id, location=location, product=product_id, reference_image=reference_image_id
+        )
+        client.delete_reference_image(name=name, retry=retry, timeout=timeout, metadata=metadata)
+
+    @GoogleCloudBaseHook.catch_http_exception
+    @GoogleCloudBaseHook.fallback_to_default_project_id
+    def add_product_to_product_set(
+        self,
+        product_set_id,
+        product_id,
+        location=None,
+        project_id=None,
+        retry=None,
+        timeout=None,
+        metadata=None,
+    ):
+        """
+        For the documentation see:
+        :py:class:`~airflow.contrib.operators.gcp_vision_operator.CloudVisionAddProductToProductSetOperator`
+        """
+        client = self.get_conn()
+
+        product_name = ProductSearchClient.product_path(project_id, location, product_id)
+        product_set_name = ProductSearchClient.product_set_path(project_id, location, product_set_id)
+
+        self.log.info('Add Product[name=%s] to Product Set[name=%s]', product_name, product_set_name)
+
+        client.add_product_to_product_set(
+            name=product_set_name, product=product_name, retry=retry, timeout=timeout, metadata=metadata
+        )
+
+        self.log.info('Product added to Product Set')
+
+    @GoogleCloudBaseHook.catch_http_exception
+    @GoogleCloudBaseHook.fallback_to_default_project_id
+    def remove_product_from_product_set(
+        self,
+        product_set_id,
+        product_id,
+        location=None,
+        project_id=None,
+        retry=None,
+        timeout=None,
+        metadata=None,
+    ):
+        """
+        For the documentation see:
+        :py:class:`~airflow.contrib.operators.gcp_vision_operator.CloudVisionRemoveProductFromProductSetOperator`
+        """
+        client = self.get_conn()
+
+        product_name = ProductSearchClient.product_path(project_id, location, product_id)
+        product_set_name = ProductSearchClient.product_set_path(project_id, location, product_set_id)
+
+        self.log.info('Remove Product[name=%s] from Product Set[name=%s]', product_name, product_set_name)
+
+        client.remove_product_from_product_set(
+            name=product_set_name, product=product_name, retry=retry, timeout=timeout, metadata=metadata
+        )
+
+        self.log.info('Product removed from Product Set')
+
+    @GoogleCloudBaseHook.catch_http_exception
+    def annotate_image(self, request, retry=None, timeout=None):
```

Comment:
A nit-pick, but don't you think this method would be more elegant and concise if we got rid of all those empty lines in between?

Reply:
Other methods have similar the number of new lines .

Reply:
From PEP8:
> Use blank lines in functions, sparingly, to indicate logical sections.
https://www.python.org/dev/peps/pep-0008/#blank-lines

From Oxford dictionary:
> sparingly
> In a restricted or infrequent manner; in small quantities.

I don't think having each other line empty is using the blank lines "in an infrequent manner". The fact that this is done elsewhere is not a strong argument to me. 
I'd suggest removing the blank lines to conform with PEP8 and general programming good practices.